appveyor.yml

```diff
@@ -23,19 +23,39 @@ environment:
       TOXENV: "py36-test-deps"
       HDF5_VSVERSION: "14"
       HDF5_DIR: "C:\\hdf5\\%HDF5_VERSION%\\%HDF5_VSVERSION%"
-      HDF5_VERSION: "1.10.1"
+      HDF5_VERSION: "1.10.6"
 
     - PYTHON: "C:\\Python37"
       TOXENV: "py37-test-deps"
       HDF5_VSVERSION: "14"
       HDF5_DIR: "C:\\hdf5\\%HDF5_VERSION%\\%HDF5_VSVERSION%"
-      HDF5_VERSION: "1.10.1"
+      HDF5_VERSION: "1.10.6"
 
     - PYTHON: "C:\\Python38"
       TOXENV: "py38-test-deps"
       HDF5_VSVERSION: "14"
       HDF5_DIR: "C:\\hdf5\\%HDF5_VERSION%\\%HDF5_VSVERSION%"
-      HDF5_VERSION: "1.10.1"
+      HDF5_VERSION: "1.10.6"
+
+    ### USING HDF5 1.12 ###
+
+    - PYTHON: "C:\\Python36"
+      TOXENV: "py36-test-deps"
+      HDF5_VSVERSION: "14"
+      HDF5_DIR: "C:\\hdf5\\%HDF5_VERSION%\\%HDF5_VSVERSION%"
+      HDF5_VERSION: "1.12.0"
+
+    - PYTHON: "C:\\Python37"
+      TOXENV: "py37-test-deps"
+      HDF5_VSVERSION: "14"
+      HDF5_DIR: "C:\\hdf5\\%HDF5_VERSION%\\%HDF5_VSVERSION%"
+      HDF5_VERSION: "1.12.0"
+
+    - PYTHON: "C:\\Python38"
+      TOXENV: "py38-test-deps"
+      HDF5_VSVERSION: "14"
+      HDF5_DIR: "C:\\hdf5\\%HDF5_VERSION%\\%HDF5_VSVERSION%"
+      HDF5_VERSION: "1.12.0"
```

Comment:
I think Appveyor was often the last one to finish already; we may decide it's not worth running the full product of Python versions and HDF5 versions, especially as we have Windows tests on Azure too. But I'm happy to see how it goes for a while before that.